idf_build_apps/app.py

```diff
@@ -292,6 +296,35 @@ def size_json_path(self) -> t.Optional[str]:
     def config(self) -> t.Optional[str]:
         return self.config_name
 
+    def _process_override_sdkconfig_file(self):
+        self.override_sdkconfig_items = BuildAppJob.override_sdkconfig_items.copy()
+        override_sdkconfig_file_items = {}
+
+        for f in BuildAppJob.override_sdkconfig_files_name:
+            if not os.path.isabs(f):
+                f = os.path.join(self.work_dir, f)
+            if not os.path.isfile(f):
+                continue
+            with open(f) as fr:
+                for line in fr:
+                    m = self.SDKCONFIG_LINE_REGEX.match(line)
+                    if not m:
+                        continue
+                    override_sdkconfig_file_items[m.group(1)] = m.group(2)
```

Comment:
each session would have one `BuildAppJob` instance

Now your implementation would call these lines for each app.

If we move these code statements to `BuildAppJob`, like

```python
BuildAppJob:
    def __init__(
        self,
        *,
        override_sdkconfig_items,
        override_sdkconfig_filenames,
        collect_app_info: t.Optional[str] = None,
        collect_size_info: t.Optional[str] = None,
        junitxml: t.Optional[str] = None,
        **kwargs,
    ):
        super().__init__(**kwargs)
        self._junitxml = junitxml
        self._collect_app_info = collect_app_info
        self._collect_size_info = collect_size_info

        self.override_sdkconfig_file_items = your_function_here(override_sdkconfig_items, override_sdkconfig_filenames)
```

the cli passed arguments would be calculated only once.

Reply:
(detailed overriding rules listed later)

Reply:
Thanks for your review! I've read it and made some changes to simplify the sdkoverride process. Now, as a result, in the pre_find_apps stage, I prepare full items and create an override file. Then, I simply put the file name into the override_sdkconfig_file_path variable in the BuildAppJob.